commands/command_utils.py

```diff
@@ -16,26 +18,52 @@
 LEAPP_UPGRADE_FLAVOUR_SAP_HANA = 'saphana'
 LEAPP_UPGRADE_PATHS = 'upgrade_paths.json'
 
-VERSION_REGEX = re.compile(r"^([1-9]\d*)\.(\d+)$")
 
+_VersionFormat = namedtuple('VersionFormat', ('human_readable', 'regex'))
 
-def check_version(version):
+
+class VersionFormats(Enum):
+    MAJOR_ONLY = _VersionFormat('MAJOR_VER', re.compile(r'^[1-9]\d*$'))
+    MAJOR_MINOR = _VersionFormat('MAJOR_VER.MINOR_VER', re.compile(r"^([1-9]\d*)\.(\d+)$"))
+
+
+class _VersionKind(str, Enum):
+    """ Enum encoding information whether the given OS version is source or target. """
+    SOURCE = 'source'
+    TARGET = 'target'
+
+
+class RHELDistros(str, Enum):
```

Comment:
I would not use the `RHELDistros` name in this context. Let's use rather something like `DistroIDs`

Reply:
Fixed, thanks.